testing/resource.go

```diff
@@ -65,11 +66,34 @@ func (c *Resource) SetDefaults(ctx context.Context) {
 	c.Spec.SetDefaults(ctx)
 }
 
+type onContextKey struct{}
+
+// WithContext is the type of a context decorator.
+type WithContext func(context.Context) context.Context
+
+// WithValue returns a WithContext for attaching an OnContext with the given value.
+func WithValue(val string) WithContext {
```

Comment:
I have found it useful to have methods that do:

```
ctx := foo.WithBar(ctx, barbarbar)
```

The usage is inline and then it is passed down the the call chain.

Reply:
Done